service/src/main/kotlin/xyz/lebkuchenfm/domain/xsounds/XSoundsService.kt

```diff
@@ -13,11 +17,16 @@ class XSoundsService(private val repository: XSoundsRepository, private val file
         return repository.findAllByTagOrderByNameAsc(tag)
     }
 
-    suspend fun addNewXSound(soundName: String, tags: List<String>, bytes: ByteArray): XSound {
-        val fileUrl = fileRepository.uploadXSoundFile(soundName, bytes)
+    sealed interface NewSoundError {
+        data object CouldNotAddNewSound : NewSoundError
+    }
+
+    suspend fun addNewXSound(soundName: String, tags: List<String>, bytes: ByteArray): Result<XSound, NewSoundError> {
+        val fileUrl = fileRepository.uploadXSoundFile(soundName, bytes).get()
+            ?: return Err(NewSoundError.CouldNotAddNewSound)
```

Comment:
This one should be about not being able to save the file into the file storage.
```suggestion
            ?: return Err(NewSoundError.CouldNotSaveFileToRepository)
```